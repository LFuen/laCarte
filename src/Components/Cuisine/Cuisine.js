import React, {Component} from 'react'
import LaCarteContext from '../../context/LaCarteContext'
import {Link} from 'react-router-dom'
import './Cuisine.css'


class Cuisine extends Component{

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = LaCarteContext

    render(){

        const {meals, cuisines} = this.context
        const originParameter = this.props.match.params.origin
        const cuisineOrg = cuisines.length > 0 ? cuisines.find(cuisine => cuisine.origin === originParameter) : ''
        const cuisineMeals = meals.filter(meal => meal.cuisine_id === cuisineOrg.id)

        if(cuisineOrg && cuisineMeals.length > 0){
        return(
            <div className='dishes'>
                <h1 className='cuisineName' key={cuisineOrg}>{cuisineOrg.origin} Dishes</h1>
                {cuisineMeals.map(meal => {
                    return (
                    <div key={meal.id}>
                        <Link to={`/meals/${meal.id}`}>
                            <h4>{meal.meal_name}</h4>
                        </Link>
                    </div>
                )})}
            </div>
        )
        } else {
            return (<h3>Sorry, we don't offer that dish just yet!</h3>)
        }
    }
}


export default Cuisine