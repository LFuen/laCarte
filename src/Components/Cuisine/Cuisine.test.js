import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import {BrowserRouter} from 'react-router-dom'
import LaCarteContext from '../../context/LaCarteContext'
import Cuisine from './Cuisine'



describe(`Cuisine Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const cuisine = renderer.create(<BrowserRouter><Cuisine/></BrowserRouter>)
                expect(cuisine).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <Cuisine />
                    </LaCarteContext.Provider>                
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})