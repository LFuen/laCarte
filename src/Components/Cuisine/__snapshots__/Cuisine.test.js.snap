// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cuisine Component Snapshot renders the UI as expected 1`] = `
<h3>
  Sorry, we don't offer that dish just yet!
</h3>
`;
